crates/ntx-builder/src/state/mod.rs

```diff
@@ -336,14 +342,15 @@ impl State {
         }
         for note in network_notes {
             tx_impact.notes.insert(note.nullifier());
-            self.nullifier_idx.insert(note.nullifier(), note.account_prefix());
-            // Skip notes which target a non-existent network account.
-            if let Some(account) = self
-                .fetch_account(note.account_prefix())
-                .await
-                .context("failed to load account")?
-            {
-                account.add_note(note);
+            // TODO: support multi target network notes.
+            if let Some(prefix) = note.account_prefix() {
+                self.nullifier_idx.insert(note.nullifier(), prefix);
+                // Skip notes which target a non-existent network account.
+                if let Some(account) =
+                    self.fetch_account(prefix).await.context("failed to load account")?
+                {
+                    account.add_note(note);
+                }
```

Comment:
Here too I would replace the `TODO` with a simple comment about not handling multi-target notes yet (mostly because enabling support for multi-target notes is a much bigger refactoring than a simple `TODO` here implies).

Also, I would expand the comment and explain why there could be non-existent network accounts.